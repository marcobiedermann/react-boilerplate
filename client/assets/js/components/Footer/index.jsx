import PropTypes from 'prop-types';
import React from 'react';

const Footer = props => {
  const { children } = props;

  return <footer>{children}</footer>;
};

Footer.propTypes = {
  children: PropTypes.node,
};

Footer.defaultProps = {
  children: null,
};

export default Footer;
