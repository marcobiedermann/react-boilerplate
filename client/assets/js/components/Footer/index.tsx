import React from 'react';

const Footer: React.FC = (props) => {
  const { children } = props;

  return <footer>{children}</footer>;
};

export default Footer;
