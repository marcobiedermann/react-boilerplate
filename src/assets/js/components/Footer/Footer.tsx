import { ReactNode } from 'react';

interface FooterProps {
  children: ReactNode;
}

function Footer(props: FooterProps): JSX.Element {
  return <footer {...props} />;
}

export type { FooterProps };
export default Footer;
