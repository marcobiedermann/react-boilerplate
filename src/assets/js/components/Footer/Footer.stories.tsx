import type { Meta, StoryObj } from '@storybook/react';
import Footer from '.';

const meta: Meta<typeof Footer> = {
  component: Footer,
};

type Story = StoryObj<typeof Footer>;

const Default: Story = {
  args: {
    children: 'Footer',
  },
};

export { Default };
export default meta;
