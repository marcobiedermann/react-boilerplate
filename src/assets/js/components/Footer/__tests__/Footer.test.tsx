/* eslint-disable import/no-extraneous-dependencies */

import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';
import * as stories from '../Footer.stories';

const { Default } = composeStories(stories);

describe('footer', () => {
  it('should render correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
