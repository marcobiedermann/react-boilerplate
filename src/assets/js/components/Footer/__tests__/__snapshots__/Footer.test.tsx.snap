// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer should render correctly 1`] = `
<footer>
  Footer
</footer>
`;
